bin/ch/ch.cpp

```diff
@@ -671,6 +671,10 @@ HRESULT ExecuteTest(const char* fileName)
         }
     }
 Error:
+    if( fileContents != nullptr)
+    {
+        free((char*)fileContents);
```

Comment:
> a deferred source initializer; even if we were to add the finalize callback for it, it doesn't handle the case where it's never needed, where we'd still need to free the data that would be needed in the case that the deferred initializer were to run

In that case I'd assume that data's lifetime is attached to the serialized bytecode buffer's lifetime. When the serialized buffer is being released, the deferred source initializer data can be released. I can try manage the data in JsCreateExternalArrayBuffer's callBackState.